### DISCUSSION TITLE
sequelize dynamic table names

### DISCUSSION THREAD

**Discussion by TSchaefer78** [sequelize/sequelize#14997] [Help & Questions]
Hello,

I have a question that Google can't answer.

Is it possible to create a dynamic table name in the model using Sequelize?

Or how can I pass a variable to the method?

Here is my code for better understanding:

regards
Tom

---------------------------------------------------------------------------------------------------------
Sequelize - Model

import { Sequelize } from "sequelize";
import db from '../config/Database.js';

const Maschines = db.define('workplaces', {

    [machineId]:{ // <-- i want to insert the passed variable here
        type: DataTypes.INTEGER,
        allowNull: false,
        validate:{
            notEmpty: true
        }
    }
}, {
    freezeTableName: true
});

export default Maschines ;

-----------------------------------------------------------------------------------------------------------
import Test from "../models/TestModel.js";

export const createMachine = async (req, res) => {
  const countMachines = 3;
  
   try {
          for (var i = 0; i < countMachines ; i++){
              await Test.create({
                  [i]: "Test"
              },i)
          }   ;
          res.status(201).json({msg: "machine created successfully"})
      } catch (error) {
          res.status(500).json({msg: error.message});
      }
  }

**Comment by fzn0x** [comment]
You can do that by creating a function that passing the different tables name to the same model structure.

**Comment by TSchaefer78** [comment]
Would you be so kind as to provide me with example code on this?
